src/test/java/BaseTest.java

```diff
@@ -1,11 +1,100 @@
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Parameters;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.openqa.selenium.interactions.Actions;
+
+
+import java.time.Duration;
+import java.util.UUID;
 
 public class BaseTest {
+    public static WebDriverWait wait;
+    public static WebDriver driver = null;
+
+    public static String url = "";
+
+    public Actions actions= null;
 
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
-}
\ No newline at end of file
+
+    @BeforeMethod
+    @Parameters({"BaseURL"})
+    public void launchBrowser(String BaseURL) {
+        //      Added ChromeOptions argument below to fix websocket error
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+        driver = new ChromeDriver(options);
+
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+
+        actions = new Actions(driver);
+        wait = new WebDriverWait(driver,Duration.ofSeconds(4));
+
+        url= BaseURL;
+        navigateToPage();
+    }
+
+    @AfterMethod
+    public void closeBrowser() {
+        driver.quit();
+    }
+
+    public static void navigateToPage() {
+        driver.get(url);
+    }
+
+
+    public static void provideEmail(String email) {
+        WebElement emailField = driver.findElement(By.cssSelector("input[type='email']"));
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+
+    public static void providePassword(String password) {
+        WebElement passwordField = driver.findElement(By.cssSelector("input[type='password']"));
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+
+    public static void clickSubmit() {
+        WebElement submit = driver.findElement(By.cssSelector("button[type='submit']"));
+        submit.click();
+    }
+
+    public static void clickSaveButton() {
+        WebElement saveButton = driver.findElement(By.cssSelector("button.btn-submit"));
+        saveButton.click();
+    }
+
+    public static void provideProfileName(String randomName) {
+        WebElement profileName = driver.findElement(By.cssSelector("[name='name']"));
+        profileName.clear();
+        profileName.sendKeys(randomName);
+    }
+
+    public static void provideCurrentPassword(String password) {
+        WebElement currentPassword = driver.findElement(By.cssSelector("[name='current_password']"));
+        currentPassword.clear();
+        currentPassword.sendKeys(password);
+    }
+
+    public static String generateRandomName() {
+        return UUID.randomUUID().toString().replace("-", "");
+    }
+
+    public static void clickAvatarIcon() {
+        WebElement avatarIcon = driver.findElement(By.cssSelector("img.avatar"));
+        avatarIcon.click();
```

Comment:
Page Object Model (POM): recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

Move these methods to the corresponding class according to the POM.

![Screenshot_2](https://github.com/CourtMatt/A47-QA-Automation-Framework/assets/104761482/96328b07-74f6-4a00-8ea8-688f6810571c)